test/libsolidity/smtCheckerTests/abi/abi_encode_no_arguments.sol

```diff
@@ -0,0 +1,18 @@
+pragma experimental SMTChecker;
+contract C {
+	function f() pure public { 
+		bytes memory res = abi.encode();
+		assert(res.length == 0); // should hold
+		assert(res.length > 0); // should fail
+		res = abi.encodePacked();
+		assert(res.length == 0); // should hold
+		assert(res.length > 0); // should fail
+		abi.encodeWithSelector(0);
+		abi.encodeWithSignature("");
```

Comment:
Can we assert something about the return value of these? If not, I will remove them.

Reply:
The first one should be `hex"00000000"`. The second should be `keccak256("")[0:4]`. Not sure the SMTEncoder can devise all these values though.

Reply:
But their length should be 4 bytes both in both cases, which the SMTEncoder should be able to handle.

Reply:
I don't think it is right now.

Reply:
The only thing it is capable of telling that two calls with the same inputs should equal in output (as is able to track the variables leading up to the call), correct?

Reply:
Exactly

Reply:
I guess we can leave a comment and a failing assertion here (`assert(res.length == 4) // should hold (if we implement support)`), so if behaviour changes it is detected.

Reply:
That is exactly what I am about to push :)